tests/test_coordinates/test_transforms.py

```diff
@@ -0,0 +1,145 @@
+    _itrs_to_wgs84,
+    _wgs84_to_itrs,
+    _itrs_to_azel, _gcrs_to_lvlh
+)
+from celest import units as u
+from unittest import TestCase
+import numpy as np
+
+
+class TestCoordinateTransformations(TestCase):
+
+    def setUp(self):
+        file_name = "tests/test_data/coordinate_validation_long.txt"
+        cols = (0, 5, 6, 7, 11, 12, 13, 14, 15, 16, 17, 18, 19)
+        skiprows = 1
+        max_rows = 5000
+        data = np.loadtxt(fname=file_name, usecols=cols, skiprows=skiprows,
+                          max_rows=max_rows)
+
+        julian = data[:, 0] + 2430000
+        wgs84 = data[:, 1:3]
+        altitude = data[:, 3]
+        gcrs = data[:, 4:7]
+        gcrs_velocity = data[:, 7:10]
+        itrs = data[:, 10:]
+
+        self.wgs84 = WGS84(julian, wgs84[:, 0], wgs84[:, 1], altitude, u.deg,
+                           u.km)
+        self.gcrs = GCRS(julian, gcrs[:, 0], gcrs[:, 1], gcrs[:, 2], u.km)
+        self.gcrs_velocity = GCRS(julian, gcrs_velocity[:, 0],
+                                  gcrs_velocity[:, 1], gcrs_velocity[:, 2],
+                                  u.m / u.s)
+        self.itrs = ITRS(julian, itrs[:, 0], itrs[:, 1], itrs[:, 2], u.km)
+
+        latitude, longitude, height = 52.1579, -106.6702, 0.482
+        self.location = GroundLocation(latitude, longitude, height, u.deg, u.km)
+
+    def test_gcrs_to_itrs_raises_error_for_invalid_input_frame(self):
+        self.assertRaises(ValueError, _gcrs_to_itrs, self.itrs)
+
+    def test_gcrs_to_itrs_raises_error_for_mismatched_dimension_units(self):
+        self.gcrs.x._unit = u.m
+        self.assertRaises(ValueError, _gcrs_to_itrs, self.gcrs)
+
+    def test_gcrs_to_itrs_for_validation(self):
+        itrs = _gcrs_to_itrs(self.gcrs)
+
+        self.assertIsInstance(itrs, ITRS)
+        self.assertTrue(np.allclose(itrs.x.data, self.itrs.x.data, atol=0.35))
+        self.assertTrue(np.allclose(itrs.y.data, self.itrs.y.data, atol=0.35))
+        self.assertTrue(np.allclose(itrs.z.data, self.itrs.z.data, atol=0.35))
+
+    def test_itrs_to_gcrs_raises_error_for_invalid_input_frame(self):
+        self.assertRaises(ValueError, _itrs_to_gcrs, self.gcrs)
+
+    def test_itrs_to_gcrs_raises_error_for_mismatched_dimension_units(self):
+        self.itrs.x._unit = u.m
+        self.assertRaises(ValueError, _itrs_to_gcrs, self.itrs)
+
+    def test_itrs_to_gcrs_for_validation(self):
+        gcrs = _itrs_to_gcrs(self.itrs)
+
+        self.assertTrue(np.allclose(gcrs.x.data, self.gcrs.x.data, atol=0.35))
+        self.assertTrue(np.allclose(gcrs.y.data, self.gcrs.y.data, atol=0.35))
+        self.assertTrue(np.allclose(gcrs.z.data, self.gcrs.z.data, atol=0.35))
+
+    def test_itrs_to_wgs84_raises_error_for_invalid_input_frame(self):
+        self.assertRaises(ValueError, _itrs_to_wgs84, self.gcrs)
+
+    def test_itrs_to_wgs84_for_validation(self):
+        wgs84 = _itrs_to_wgs84(self.itrs)
+
+        self.assertTrue(np.allclose(wgs84.latitude.to(u.deg).data,
```

Comment:
Move towards black format. Also the `.data` should throw an error; explore this.